### PR TITLE
[OF-1228] feat: Added GetShopifyStores

### PR DESCRIPTION
**Pull request by MAG-AW**
* Added new API for retrieving stores for a given user
* Added a manual test and verified as working

connected-spaces-platform Pull Request

**For the reviewer**

* [ ] If required, are the changes covered by appropriate tests?
* [ ] Are any public-facing API changes well documented?
* [ ] Is the code easily readable and extensible and/or follow existing conventions?

### CODE REVIEW

File: Library/include/CSP/Systems/ECommerce/ECommerceSystem.h [hunk 1]
```diff
@@ -94,6 +94,11 @@ class CSP_API ECommerceSystem : public SystemBase
 	/// @param Callback CartInfoResultCallback : Callback when asynchronous task finishes
 	CSP_ASYNC_RESULT void GetCart(const common::String& SpaceId, const common::String& CartId, CartInfoResultCallback Callback);
 
+	/// @brief Gets all shopify stores for the given user.
+	/// @param UserId csp::common::String : ID of the user to get all stores for.
+	/// @param Callback GetShopifyStoresResultCallback : Callback when asynchronous task finishes
+	CSP_ASYNC_RESULT void GetShopifyStores(const common::String& UserId, GetShopifyStoresResultCallback Callback);
```
Comment (MAG-AdamThorn) [inline Library/include/CSP/Systems/ECommerce/ECommerceSystem.h:100]: Can you actually retrieve stores for other users?

File: Tests/src/PublicAPITests/ECommerceSystemTests.cpp [hunk 2]
```diff
@@ -807,4 +807,55 @@ CSP_PUBLIC_TEST(CSPEngine, ECommerceSystemTests, AddShopifyStoreTest)
...
+	EXPECT_EQ(ShopifyStore.StoreName, StoreName);
+
+	auto [GetShopifyStoresResult] = AWAIT_PRE(ECommerceSystem, GetShopifyStores, RequestPredicate, UserId);
+
+	EXPECT_EQ(GetShopifyStoresResult.GetShopifyStores()[0].StoreId, ShopifyStore.StoreId);
+	EXPECT_EQ(GetShopifyStoresResult.GetShopifyStores()[0].SpaceId, ShopifyStore.SpaceId);
+	EXPECT_EQ(GetShopifyStoresResult.GetShopifyStores()[0].SpaceOwnerId, ShopifyStore.SpaceOwnerId);
+	EXPECT_EQ(GetShopifyStoresResult.GetShopifyStores()[0].StoreName, ShopifyStore.StoreName);
+
+	LogOut(UserSystem);
```
Comment (MAG-AdamThorn) [inline Tests/src/PublicAPITests/ECommerceSystemTests.cpp:859]: Should we be removing the store we added at the end of the test?

**Comment by MAG-ME** [comment]
It might be useful to expose the option to restrict the query to just return active stores

**Comment by MAG-SamBirley** [comment]
> It might be useful to expose the option to restrict the query to just return active stores

What's the LOE on that @MAG-AW? If it's easy we might as well do it whilst in the area.

**Review by MAG-AdamThorn — DISMISSED** [review]

**Review by MAG-SamBirley — APPROVED** [review]
Love it, nice work! cc @MAG-ME @MAG-DanielMurdolo - you're getting the optional `Active` param 🎉

File: Library/src/Systems/ECommerce/ECommerceSystem.cpp [hunk 3]
```diff
@@ -126,6 +126,28 @@ void ECommerceSystem::GetCart(const common::String& SpaceId, const common::Strin
...
+{
+	csp::services::ResponseHandlerPtr ResponseHandler
+		= ShopifyAPI->CreateHandler<GetShopifyStoresResultCallback, GetShopifyStoresResult, void, csp::services::DtoArray<chs::ShopifyStorefrontDto>>(
+			Callback,
+			nullptr,
+			csp::web::EResponseCodes::ResponseCreated);
+
+	std::optional<bool> ActiveParam;
+
+    if (IsActive.HasValue())
```
Comment (MAG-SamBirley) [inline Library/src/Systems/ECommerce/ECommerceSystem.cpp:139]: Not a reason to block this PR, but just as a note for the future - the logic here is equivalent to just passing `IsActive` straight through to the endpoint call, no need for a local optional variable here.

**Pull request merged**